### PR TITLE
HIVE-21766 : Select * returns no rows in hive bootstrap from a statior dynamic partitioned managed table with Timestamp type as partition column from on prem to WASB even though count ( * ) matches

### PR DESCRIPTION
**Pull request by maheshk114**
…

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/exec/ReplTxnTask.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ public int execute(DriverContext driverContext) {
     }
 
     try {
-      HiveTxnManager txnManager = driverContext.getCtx().getHiveTxnManager();
+      HiveTxnManager txnManager = driverContext.getReplTxnManager(conf);
```
Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/ReplTxnTask.java]: Noticed in several places, SessionState.get().getTxnMgr() is used to get txnmgr on which performed some operations. For example, addWriteNotificationLog, Hive.getTable etc.
I think, those operations would be pointing to source metastore. Might go wrong.

File: ql/src/java/org/apache/hadoop/hive/ql/DriverContext.java [hunk 2]
```diff
@@ -234,4 +240,24 @@ public void apply(FileSinkOperator fsOp) {
...
+  // Create a transaction manager for replication flow. In case user has provided a different config for REPL load,
+  // this transaction manager will be used to do the transaction operation. This is specifically useful when REPL load
+  // is called with different hive metastore uri expecting the transaction information to be stored in the specified
+  // metastore.
+  public synchronized HiveTxnManager getReplTxnManager(HiveConf hiveConf) throws LockException {
+    if (replTxnManager == null) {
+      replTxnManager = TxnManagerFactory.getTxnManagerFactory().getTxnManager(hiveConf);
+      LOG.debug("Created a new replTxnManager : " + replTxnManager);
+    }
+    return replTxnManager;
```
[the remark this answers is not in the document]
Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/DriverContext.java]: that scenario is not applicable as for one repl command only one txn manager will be created and we can don't support config change in between.

File: ql/src/java/org/apache/hadoop/hive/ql/exec/ReplTxnTask.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/ReplTxnTask.java]: This is required for migration case also. I feel its better to keep it for all as cost of creating txn manager is not much and this way the same flow will be there for all kind. Easy to maintain.

Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/ReplTxnTask.java]: yes ..the changes looks risky ..

**Pull request closed without merging**